hkmc2/shared/src/test/mlscript-compile/apps/parsing/ParseRule.mls

```diff
@@ -124,19 +124,16 @@ fun rule(name, ...choices) = ParseRule(name, choices Iter.toStack())
 
 module Choice with
   class Keyword(keyword: Keyword, rest: ParseRule)
-  class Expr(process: Tree -> Tree, rest: ParseRule)
-  class Type(process: Tree -> Tree, rest: ParseRule)
-  class Terminal(process: (Token, Tree) -> Tree, token: Token, rest: ParseRule)
+  class Expr(isType: Bool, process: Tree -> Tree, rest: ParseRule)
```

Comment:
Any reason for this design? Having separate Expr and Type seems cleaner, especially when we generalize the approach to languages with more categories (and possibly some added dynamically in the program itself).